tests/MongoDB.Driver.Examples/ChangeStreamExamples.cs

```diff
@@ -122,5 +126,90 @@ public void ChangeStreamExample3()
                 next.FullDocument.Should().Be(documents[1]);
             }
         }
+
+        [Fact]
+        public void Changestream_Example_4()
+        {
+            RequireServer.Check().Supports(Feature.AggregateAddFields);
+
+            var client = DriverTestConfiguration.Client;
+            var database = client.GetDatabase("ChangeStreamExamples");
+            database.DropCollection("inventory");
+            var inventoryCollection = database.GetCollection<BsonDocument>("inventory");
+
+            //const pipeline = [
+            //{ $match: { $or : [{ 'fullDocument.username': 'alice' }, { 'operationType' : 'delete' }] } },
+
+            //{ $addFields: { newField: 'this is an added field!' } }
+
+            //];
+            //const collection = db.collection('inventory');
+            //const changeStream = collection.watch(pipeline);
+            //const next = await changeStream.next();
+
+            var cancelationTokenSource = new CancellationTokenSource();
+            try
+            {
+                BsonDocument document = new BsonDocument("username", "alice");
+
+                Task.Run(() =>
+                {
+                    while (!cancelationTokenSource.IsCancellationRequested)
+                    {
+                        Thread.Sleep(TimeSpan.FromMilliseconds(500));
+                        inventoryCollection.InsertOne(document);
+                        inventoryCollection.DeleteOne(document);
+                    }
+                }, cancelationTokenSource.Token);
+
+                // Start Changestream Example 4
+                var pipeline = new EmptyPipelineDefinition<ChangeStreamDocument<BsonDocument>>()
+                    .Match(filter =>
+                        filter.FullDocument["username"] == "alice" ||
+                        filter.OperationType == ChangeStreamOperationType.Delete)
+                    .AppendStage(
+                        new BsonDocumentPipelineStageDefinition<ChangeStreamDocument<BsonDocument>, BsonDocument>(
+                            new BsonDocument
+                            {
+                                {"$addFields", new BsonDocument("newField", "this is an added field!")}
+                            }));
+
+                var collection = database.GetCollection<BsonDocument>("inventory");
+                var changeStream = collection.Watch(pipeline);
+
+                var enumerator = changeStream.ToEnumerable().GetEnumerator();
+                enumerator.MoveNext();
```

Comment:
The ticket just asks for one change event, so let's match that expectation so the examples are consistent across all driver examples.

Reply:
MoveNext returns a bool which this sample code is ignoring.

I know the pseudo-code doesn't require us to test the return value of MoveNext, but I'm concerned that as a documentation example it is misleading to not test it. In real applications and application MUST test the return value of MoveNext.

Reply:
Why was this conversation "Resolved"? The question is still pending.

Reply:
I've added if construction to check moveNext() result